From: Chun Han Hsiao <chhsiao90@gmail.com>
Date: Mon, 7 Mar 2022 16:24:04 +0800
Subject: [PATCH] Fix wrong options while using async-profiler v2

Previously, the jmh will call async-profiler with output=flamegraph,
which the 'output' option is not a valid option for both async-profiler
v1 and v2.

For jfr format, this class will call the profiler with
'start,file=name.jfr', async-profiler can get the output format from the
extension corretly, so we don't need to have a redundant option like
'start,jfr,file=name.jfr'.

For other formats like flamegraph, this class call the dump separately
with the format option included in the option, like
`flamegraph,file=dump.html`.
---
 .../src/main/java/org/openjdk/jmh/profile/AsyncProfiler.java     | 1 -
 1 file changed, 1 deletion(-)

diff --git jmh-core/src/main/java/org/openjdk/jmh/profile/AsyncProfiler.java jmh-core/src/main/java/org/openjdk/jmh/profile/AsyncProfiler.java
index e04922ba..084393cf 100644
--- jmh-core/src/main/java/org/openjdk/jmh/profile/AsyncProfiler.java
+++ jmh-core/src/main/java/org/openjdk/jmh/profile/AsyncProfiler.java
@@ -251,7 +251,6 @@ public final class AsyncProfiler implements ExternalProfiler, InternalProfiler {
             direction = optDirection.value(set);
 
             output = optOutput.values(set);
-            builder.appendMulti(optOutput);
 
             // Secondary events are those that may be collected simultaneously with a primary event in a JFR profile.
             // To be used as such, we require they are specifed with the lock and alloc option, rather than event=lock,
-- 
2.35.1

